pkg/collector/corechecks/ebpf/probe/ebpfcheck/names.go

```diff
@@ -0,0 +1,61 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+package ebpfcheck
+
+import (
+	manager "github.com/DataDog/ebpf-manager"
+	"github.com/cilium/ebpf"
+)
+
+var mapNameMapping = make(map[uint32]string)
+
+// AddNameMappings adds the full name mappings for ebpf maps in the manager
+func AddNameMappings(mgr *manager.Manager) {
+	maps, err := mgr.GetMaps()
+	if err != nil {
+		return
+	}
+	iterateMaps(maps, func(mapid uint32, name string) {
+		mapNameMapping[mapid] = name
+	})
+}
+
+// AddNameMappingsCollection adds the full name mappings for ebpf maps in the collection
+func AddNameMappingsCollection(coll *ebpf.Collection) {
+	iterateMaps(coll.Maps, func(mapid uint32, name string) {
+		mapNameMapping[mapid] = name
+	})
+}
+
+// RemoveNameMappings removes the full name mappings for ebpf maps in the manager
+func RemoveNameMappings(mgr *manager.Manager) {
```

Comment:
Maybe I'm missing something obvious, but why can't the manager do this automatically instead of requiring each module to call AddNameMappings?

Reply:
Call AddNameMappingsCollection as part of manager.NewMap and RemoveNameMappings as part of manager.Stop